Sources/Load.php

```diff
@@ -656,9 +658,9 @@ function loadUserSettings()
 		{
 			$tfa_data = $smcFunc['json_decode']($_COOKIE[$cookiename . '_tfa'], true);
 
-			list ($id, $user, $exp, $domain, $path, $preserve) = $tfa_data;
+			list ($id, $user, $exp, $domain, $path) = array_pad((array) $tfa_data, 5, '');
```

Comment:
Since the only variable that is actually used is $exp, can we not refactor this?  Clean up various parts of the codebase are touched.